### PR TITLE
fs/littlefs: improve littlefs flash block erase balance performance

### PR DESCRIPTION
**Pull request by xiaomi-hujie**

### CODE REVIEW

File: fs/littlefs/Kconfig [hunk 1]
```diff
@@ -4,3 +4,13 @@ config FS_LITTLEFS
 	depends on !DISABLE_MOUNTPOINT
 	---help---
 		Build the LITTLEFS file system. https://github.com/ARMmbed/littlefs.
+
+config LFS_BLOCK_MUL_FACTOR
```
Comment (xiaoxiang781216) [inline fs/littlefs/Kconfig:8]: FS_LITTLEFS_BLOCK_FACTOR?

File: fs/littlefs/Kconfig [hunk 2]
```diff
@@ -4,3 +4,13 @@ config FS_LITTLEFS
...
+	int "LITTLEFS Block size multiple factor"
+	default 4
+	depends on FS_LITTLEFS
+
+config LFS_BLOCK_CYCLE
```
Comment (xiaoxiang781216) [inline fs/littlefs/Kconfig:13]: FS_LITTLEFS_BLOCK_CYCLE?

File: fs/littlefs/lfs_vfs.c [hunk 3]
```diff
@@ -1019,12 +1019,14 @@ static int littlefs_bind(FAR struct inode *driver, FAR const void *data,
   fs->cfg.prog           = littlefs_write_block;
   fs->cfg.erase          = littlefs_erase_block;
   fs->cfg.sync           = littlefs_sync_block;
-  fs->cfg.read_size      = fs->geo.blocksize;
+  fs->cfg.read_size      = fs->geo.blocksize \
```
Comment (xiaoxiang781216) [inline fs/littlefs/lfs_vfs.c:1022]: remove \

File: fs/littlefs/lfs_vfs.c [hunk 4]
```diff
@@ -1019,12 +1019,14 @@ static int littlefs_bind(FAR struct inode *driver, FAR const void *data,
...
+                           * CONFIG_FS_LITTLEFS_BLOCK_FACTOR;
   fs->cfg.prog_size      = fs->geo.blocksize;
   fs->cfg.block_size     = fs->geo.erasesize;
   fs->cfg.block_count    = fs->geo.neraseblocks;
-  fs->cfg.block_cycles   = 500;
-  fs->cfg.cache_size     = fs->geo.blocksize;
+  fs->cfg.block_cycles   = CONFIG_FS_LITTLEFS_BLOCK_CYCLE;
+  fs->cfg.cache_size     = fs->geo.blocksize \
```
Comment (xiaoxiang781216) [inline fs/littlefs/lfs_vfs.c:1028]: remove \

File: arch/risc-v/src/esp32c3/esp32c3_spiflash.c [hunk 5]
```diff
@@ -72,7 +72,7 @@
 #define MMU_BYTES2PAGES(_n)         (((_n) + SPI_FLASH_MMU_PAGE_SIZE - 1) / \
                                      SPI_FLASH_MMU_PAGE_SIZE)
 
-#define SPI_FLASH_BLK_SIZE          256
+#define SPI_FLASH_BLK_SIZE          CONFIG_ESP32C3_SPIFLASH_BLOCK_SIZE
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/esp32c3/esp32c3_spiflash.c:75]: let's split the esp32 change to another patch

**Comment by Ouss4** [comment]
@xiaomi-hujie @xiaoxiang781216 is it possible to have a proper email address for these commits (it seems to be fixed to a template)?  Would make life easier later if we needed to identify the author.

File: fs/littlefs/lfs_vfs.c [hunk 6]
```diff
@@ -1019,12 +1019,14 @@ static int littlefs_bind(FAR struct inode *driver, FAR const void *data,
   fs->cfg.prog           = littlefs_write_block;
   fs->cfg.erase          = littlefs_erase_block;
   fs->cfg.sync           = littlefs_sync_block;
-  fs->cfg.read_size      = fs->geo.blocksize;
+  fs->cfg.read_size      = fs->geo.blocksize
+                           * CONFIG_FS_LITTLEFS_BLOCK_FACTOR;
```
Comment (gustavonihei) [inline fs/littlefs/lfs_vfs.c:1023]:
```suggestion
  fs->cfg.read_size      = fs->geo.blocksize *
                           CONFIG_FS_LITTLEFS_BLOCK_FACTOR;
```
I think there is no entry on the coding standard for this, but the most usual is to break the line after the operator.

File: fs/littlefs/lfs_vfs.c [hunk 7]
```diff
@@ -1019,12 +1019,14 @@ static int littlefs_bind(FAR struct inode *driver, FAR const void *data,
...
   fs->cfg.prog_size      = fs->geo.blocksize;
   fs->cfg.block_size     = fs->geo.erasesize;
   fs->cfg.block_count    = fs->geo.neraseblocks;
-  fs->cfg.block_cycles   = 500;
-  fs->cfg.cache_size     = fs->geo.blocksize;
+  fs->cfg.block_cycles   = CONFIG_FS_LITTLEFS_BLOCK_CYCLE;
+  fs->cfg.cache_size     = fs->geo.blocksize
+                           * CONFIG_FS_LITTLEFS_BLOCK_FACTOR;
```
Comment (gustavonihei) [inline fs/littlefs/lfs_vfs.c:1029]:
```suggestion
  fs->cfg.cache_size     = fs->geo.blocksize *
                           CONFIG_FS_LITTLEFS_BLOCK_FACTOR;
```

File: fs/littlefs/Kconfig [hunk 8]
```diff
@@ -4,3 +4,13 @@ config FS_LITTLEFS
...
 		Build the LITTLEFS file system. https://github.com/ARMmbed/littlefs.
+
+config FS_LITTLEFS_BLOCK_FACTOR
+	int "LITTLEFS Block size multiple factor"
+	default 4
+	depends on FS_LITTLEFS
+
+config FS_LITTLEFS_BLOCK_CYCLE
+	int "LITTLEFS Block Cycle"
+	default 200
```
Comment (gustavonihei) [inline fs/littlefs/Kconfig:15]: Please, contribute with a help text for both configs.

File: fs/littlefs/Kconfig [hunk 9]
```diff
@@ -4,3 +4,13 @@ config FS_LITTLEFS
 	depends on !DISABLE_MOUNTPOINT
 	---help---
 		Build the LITTLEFS file system. https://github.com/ARMmbed/littlefs.
+
+config FS_LITTLEFS_BLOCK_FACTOR
+	int "LITTLEFS Block size multiple factor"
+	default 4
+	depends on FS_LITTLEFS
+
+config FS_LITTLEFS_BLOCK_CYCLE
```
Comment (gustavonihei) [inline fs/littlefs/Kconfig:13]: This config also depends on `FS_LITTLEFS`.
Wrap these two in a outer `if/endif` block to avoid spreading `depends on` statements.

File: fs/littlefs/Kconfig [hunk 10]
```diff
@@ -4,3 +4,13 @@ config FS_LITTLEFS
 	depends on !DISABLE_MOUNTPOINT
 	---help---
 		Build the LITTLEFS file system. https://github.com/ARMmbed/littlefs.
```
Comment (gustavonihei) [inline fs/littlefs/Kconfig:6]:
```suggestion
		Build the LITTLEFS file system. https://github.com/littlefs-project/littlefs.
```
Please, could you also update the project URL?

**Review by xiaoxiang781216 — APPROVED** [review]
